deploy/manifests/nginx-gateway.yaml

```diff
@@ -84,62 +84,25 @@ spec:
       labels:
         app: nginx-gateway
     spec:
-      shareProcessNamespace: true
       serviceAccountName: nginx-gateway
       securityContext:
         sysctls:
         - name: "net.ipv4.ip_unprivileged_port_start"
           value: "0"
-      volumes:
-      - name: nginx-config
-        emptyDir: { }
-      - name: var-lib-nginx
-        emptyDir: { }
-      - name: njs-modules
-        configMap:
-          name: njs-modules
-      initContainers:
-      - image: busybox:1.34 # FIXME(pleshakov): use gateway container to init the Config with proper main config
-        name: nginx-config-initializer
-        command: [ 'sh', '-c', 'echo "load_module /usr/lib/nginx/modules/ngx_http_js_module.so; events {}  pid /etc/nginx/nginx.pid; error_log stderr debug; http { include /etc/nginx/conf.d/*.conf; js_import /usr/lib/nginx/modules/njs/httpmatches.js; }" > /etc/nginx/nginx.conf && mkdir /etc/nginx/conf.d /etc/nginx/secrets && chown 1001:0 /etc/nginx/conf.d /etc/nginx/secrets' ]
-        volumeMounts:
-        - name: nginx-config
-          mountPath: /etc/nginx
       containers:
-      - image: docker.io/nginx-kubernetes-gateway:edge # FIXME(kate-osborn): change back to ghcr before merging to main
+      - image: docker.io/library/nginx-kubernetes-gateway:edge # FIXME(kate-osborn): change back to ghcr before merging to main
         imagePullPolicy: IfNotPresent # FIXME(kate-osborn): change back to Always before merging to main
         name: nginx-gateway
         ports:
         - name: grpc
           containerPort: 54789
-        volumeMounts:
-        - name: nginx-config
-          mountPath: /etc/nginx
         securityContext:
-          runAsUser: 1001
-          # FIXME(pleshakov) - figure out which capabilities are required
-          # dropping ALL and adding only CAP_KILL doesn't work
-          # Note: CAP_KILL is needed for sending HUP signal to NGINX main process
+          runAsUser: 1001 #FIXME(kate-osborn): figure out what securityContext we need.
         args:
         - --gateway-ctlr-name=k8s-gateway.nginx.org/nginx-gateway-controller
         - --gatewayclass=nginx
-      - image: nginx:1.23 # I will remove the nginx container once the control plane can push config to nginx-with-agent.
-        imagePullPolicy: IfNotPresent
-        name: nginx
-        ports:
-        - name: http
-          containerPort: 80
-        - name: https
-          containerPort: 443
-        volumeMounts:
-        - name: nginx-config
-          mountPath: /etc/nginx
-        - name: var-lib-nginx
-          mountPath: /var/lib/nginx
-        - name: njs-modules
-          mountPath: /usr/lib/nginx/modules/njs
-      - name: nginx-with-agent
-        image: docker.io/nginx-kubernetes-gateway/nginx-with-agent:edge
+      - image: docker.io/nginx-kubernetes-gateway/nginx-with-agent:edge
```

Comment:
@pleshakov I'm not sure how much value we're getting from keeping the agent in the same Pod as the control plane. During testing, I have to test the agent running in a separate Pod to ensure the control plane can handle many agents. So, in practice, I have to test both the one Pod and many Pod cases.